.github/workflows/version-and-release.yml

```diff
@@ -0,0 +1,39 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: get version from tag
+        id: get_version
+        run: |
+          realversion="${GITHUB_REF/refs\/tags\//}"
+          realversion="${realversion//v/}"
+          echo "::set-output name=VERSION::$realversion"
+
+      - name: set version from tag
+        uses: actions/setup-java@v2
+        run: mvn versions:set -DnewVersion=${{ steps.get_version.outputs.VERSION }}
```

Comment:
@chayim should we also right after this set push version update to the pom.xml to bet `realversion+0.1'-SNAPSHOT?

Reply:
IMHO, I think that all versioning should be done in master. So as we tag, it's instantly versioned to the tag name, meaning master can go on doing whatever is in the pom.xml. The act of tagging versions here.

Reply:
> all versioning should be done in master

@chayim All versioning can't be done in master. There would be times when we'd have to maintain multiple versions. E.g. master is targeting version 2.0 and 1.6/7/8 has to be maintained/released.

Reply:
@sazzad16 why not? The versioning is basically immaterial. You can have branches and do whatever you want there to poms, etc. This approach just means that the version for pushing **the release** to maven central will be generated from the **tag**.

I'd argue this allows you to maintain as many branches and versions you want, since it's immaterial now. But, the release happens from the tag - and is guaranteed to match the tag name. The caveat is that tags with alphabetical prefixes like _v1.2.3_ become _1.2.3_.

Reply:
To be clear the versions in the pom files are never overridden or altered - except in the tag case. This means that each branch can have whatever versions and rules deemed appropriate - only on tag do we force the version to match the tag.